src/components/TextField/tests/TextField.test.tsx

```diff
@@ -595,13 +595,18 @@ describe('<TextField />', () => {
           id="MyField"
           onChange={noop}
           helpText="Help text"
+          showCharacterCount
+          maxLength={5}
         />,
       );
 
       expect(textField.find(Labelled)).toHaveLength(1);
       expect(textField.find(Labelled).prop('label')).toBe('TextField');
       expect(textField.find(Labelled).prop('id')).toBe('MyField');
       expect(textField.find(Labelled).prop('helpText')).toBe('Help text');
+      expect(textField.find(Labelled).prop('showCharacterCount')).toBe(true);
+      expect(textField.find(Labelled).prop('characterCount')).toBe(0);
+      expect(textField.find(Labelled).prop('maxLength')).toBe(5);
```

Comment:
This could also use a test for when the field's value is of length greater than zero, asserting that the character count does in fact count the characters.